### PR TITLE
OIDC authenticator defaults to using host's root CA pool if CA file is not provided

### PR DESCRIPTION
**Pull request by dimityrmirchev**
This PR proposes an alternative approach to https://github.com/brancz/kube-rbac-proxy/issues/259 by implementing https://github.com/brancz/kube-rbac-proxy/issues/259#issuecomment-2516553954.

It also aligns the behaviour of the `--oidc-ca-file` flag with [its description](https://github.com/brancz/kube-rbac-proxy/blob/28ede67b07bfa78103e78a4c4ce43560d6b15dba/cmd/kube-rbac-proxy/app/options/options.go#L126).

Reference to the code that utilises the host's root cert pool when no `CAContentProvider` is set https://github.com/kubernetes/kubernetes/blob/c83cbee114ddb732cdc06d3d1b62c9eb9220726f/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L271-L290

### CODE REVIEW

File: pkg/authn/oidc.go [hunk 1]
```diff
@@ -38,9 +38,13 @@ var (
 
 // NewOIDCAuthenticator returns OIDC authenticator
 func NewOIDCAuthenticator(ctx context.Context, config *OIDCConfig) (*OIDCAuthenticator, error) {
-	dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
-	if err != nil {
-		return nil, err
+	var dynamicCA *dynamiccertificates.DynamicFileCAContent
+	if len(config.CAFile) > 0 {
+		dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
```
Comment (stlaz) [inline pkg/authn/oidc.go:43]: nit:
```suggestion
	if len(config.CAFile) > 0 { // if unset, the OIDC authenticator defaults to host's trust store
	    var err error
		dynamicCA, err = dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
```

Comment (dimityrmirchev) [reply pkg/authn/oidc.go]: Addressed with https://github.com/brancz/kube-rbac-proxy/pull/342/commits/84cafebed9010c55eef5950ce21a4cffe2dfe9c6

Should this be squashed into the previous commit?

Comment (stlaz) [reply pkg/authn/oidc.go]: Yes please

**Comment by stlaz** [comment]
lgtm, let's wait for the tests to complete

**Comment by stlaz** [comment]
Merging, thank you for your PR :slightly_smiling_face:

**Pull request merged**